src/app/parcels/ParcelsPage.tsx

```diff
@@ -181,90 +358,221 @@ const ParcelsPage: React.FC<{}> = () => {
     const searchParams = useSearchParams();
     const parcelId = searchParams.get(parcelIdParam);
 
+    const [sortState, setSortState] = useState<SortState<ParcelsTableRow>>({ sort: false });
+
     useEffect(() => {
         if (parcelId) {
             setSelectedParcelId(parcelId);
             setModalIsOpen(true);
         }
     }, [parcelId]);
 
-    useEffect(() => {
-        (async () => {
-            setIsLoading(true);
-            const dateRangeToFetch = { ...packingDateRange };
-            const tableData = await fetchAndFormatParcelTablesData(dateRangeToFetch);
+    const [perPage, setPerPage] = useState(10);
+    const [currentPage, setCurrentPage] = useState(1);
+    const startPoint = (currentPage - 1) * perPage;
+    const endPoint = currentPage * perPage - 1;
 
-            if (areDateRangesIdentical(dateRangeToFetch, packingDateRange)) {
-                setTableData(tableData);
-            }
+    const [primaryFilters, setPrimaryFilters] = useState<Filter<ParcelsTableRow, any>[]>([]);
+    const [additionalFilters, setAdditionalFilters] = useState<Filter<ParcelsTableRow, any>[]>([]);
 
-            setIsLoading(false);
-        })();
-    }, [packingDateRange]);
+    const [areFiltersLoadingForFirstTime, setAreFiltersLoadingForFirstTime] =
```

Comment:
we need this because the filters are now built asynchronously, because the checkbox filters rely on requests to the database. We don't want to build the table until the filters are ready.